src/condor_utils/passwd_cache.unix.cpp

```diff
@@ -98,72 +98,64 @@ passwd_cache::getUseridMap(std::string &usermap)
 void
 passwd_cache::loadConfig() {
 		// initialize cache with any configured mappings
-	char *usermap_str = param("USERID_MAP");
-	if( !usermap_str ) {
+	std::string usermap_str;
+	param(usermap_str, "USERID_MAP");
+	if (usermap_str.empty()) {
 		return;
 	}
-
+	
 		// format is "username=uid,gid,gid2,gid3,... user2=uid2,gid2,..."
 		// first split on spaces, which separate the records
 		// If gid2 is "?", then we assume that supplemental groups
 		// are unknown.
-	StringList usermap(usermap_str," ");
-	free( usermap_str );
-
-	char *username;
-	usermap.rewind();
-	while( (username=usermap.next()) ) {
-		char *userids = strchr(username,'=');
-		ASSERT( userids );
-		*userids = '\0';
-		userids++;
+	for (const auto &name_equal_uid_gids: StringTokenIterator(usermap_str, " ")) {
+		size_t pos = name_equal_uid_gids.find('=');
+		ASSERT(pos != std::string::npos);
+		std::string username = name_equal_uid_gids.substr(0, pos);
+		std::string uid_gids = name_equal_uid_gids.substr(pos + 1);
 
-			// the user ids are separated by commas
-		StringList ids(userids,",");
-		ids.rewind();
+			// the user/group ids are separated by commas
+		std::vector<std::string> ids = split(uid_gids,",");
 
+		if (ids.size() < 2) {
+			EXCEPT("INVALID USERID_MAP entry %s=%s", username.c_str(), uid_gids.c_str());
+		}
 		struct passwd pwent;
-
-		char const *idstr = ids.next();
-		uid_t uid;
+		const std::string &idstr = ids.front();
+		uid_t uid ;
 		gid_t gid;
-		if( !idstr || !parseUid(idstr,&uid) ) {
-			EXCEPT("Invalid USERID_MAP entry %s=%s",username,userids);
+		if( !parseUid(idstr.c_str(),&uid) ) {
+			EXCEPT("INVALID USERID_MAP entry %s=%s", username.c_str(), uid_gids.c_str());
 		}
-		idstr = ids.next();
-		if( !idstr || !parseGid(idstr, gid) ) {
-			EXCEPT("Invalid USERID_MAP entry %s=%s",username,userids);
+		const std::string &gidstr = ids[1];
+		if(!parseGid(gidstr.c_str(), gid) ) {
+			EXCEPT("INVALID USERID_MAP entry %s=%s", username.c_str(), uid_gids.c_str());
 		}
-		pwent.pw_name = username;
+		pwent.pw_name = const_cast<char *>(username.c_str());
 		pwent.pw_uid = uid;
 		pwent.pw_gid = gid;
 		cache_uid(&pwent);
 
-		idstr = ids.next();
-		if( idstr && !strcmp(idstr,"?") ) {
+		const std::string supgidstr = ids.size() > 2 ? ids[2] : std::string("");
+		if( supgidstr == "?") {
 			continue; // no information about supplemental groups
 		}
 
-		ids.rewind();
-		ids.next(); // go to first group id
-
-		auto [it, success] = group_table.insert({username, group_entry()});
+		auto [it, success] = group_table.emplace(username, group_entry());
 		group_entry& group_cache_entry = it->second;
 
-			/* now get the group list */
-		group_cache_entry.gidlist.resize(ids.number()-1);
-
-		for (auto& gid: group_cache_entry.gidlist) {
-			idstr = ids.next();
-			ASSERT( idstr );
+			/* now get the supplemental group list */
+		group_cache_entry.gidlist.resize(ids.size()-1);
+		for (auto gid_it = ids.begin() + 2;
```

Comment:
```suggestion
		for (auto gid_it = ids.begin() + 1;
```
Upon more careful reading, I see that the original code is correct and includes all of the gids. I believe your new code needs to skip just the uid (the first item in the vector).